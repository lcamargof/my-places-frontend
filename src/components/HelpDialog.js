import React from 'react';
import PropTypes from 'prop-types';
import { Dialog, DialogContent, DialogContentText, DialogTitle, withStyles } from 'material-ui';

const styles = () => ({

});

const HelpDialog = ({ classes, open, onClose }) => (
  <Dialog onClose={onClose} open={open} aria-labelledby="help-dialog">
    <DialogTitle>Instructions</DialogTitle>
    <DialogContent>
      <DialogContentText>
        Easy instructions to use my places!
        <ul>
          <li>Right click on the map to add a place</li>
          <li>Drag the place to change their location</li>
          <li>Click on the place to know more about it</li>
          <li>When you click on a place, you can edit or remove it, do as you wish!</li>
          <li>Enjoy using myPlaces</li>
        </ul>
      </DialogContentText>
    </DialogContent>
  </Dialog>
);

HelpDialog.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default withStyles(styles)(HelpDialog)