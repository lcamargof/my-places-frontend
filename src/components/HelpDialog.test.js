import React from 'react';
import { shallow } from 'enzyme';
import HelpDialog from './HelpDialog';
import { shallowToJson } from 'enzyme-to-json';

describe('Component: HelpDialog', () => {
  it('Should render correctly', () => {
    const component = shallow(<HelpDialog open={true} onClose={jest.fn()}/>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});