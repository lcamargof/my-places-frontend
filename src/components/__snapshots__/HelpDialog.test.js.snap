// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: HelpDialog Should render correctly 1`] = `
<WithStyles(Dialog)
  aria-labelledby="help-dialog"
  onClose={[Function]}
  open={true}
>
  <WithStyles(DialogTitle)>
    Instructions
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)>
      Easy instructions to use my places!
      <ul>
        <li>
          Right click on the map to add a place
        </li>
        <li>
          Drag the place to change their location
        </li>
        <li>
          Click on the place to know more about it
        </li>
        <li>
          When you click on a place, you can edit or remove it, do as you wish!
        </li>
        <li>
          Enjoy using myPlaces
        </li>
      </ul>
    </WithStyles(DialogContentText)>
  </WithStyles(DialogContent)>
</WithStyles(Dialog)>
`;
